docs/glossary.md

```diff
@@ -0,0 +1,361 @@
+
+
+## C
+
+---
+
+#### Command Line Interface (CLI)
+
+An interface a user interacts with via the command line - often referring to a CLI wallet, which is operated using various tools and commands.
+
+#### Confirmation
+
+A transaction is confirmed when it has been included in a block that has been verified by the network (added to the blockchain). Each additional block added to the blockchain reconfirms all transactions in all previous blocks. The number of times a transaction has been confirmed is used as a measure of confidence that the transaction will remain in the blockchain. A wallet or other service may require a certain number of confirmations before it considers a transaction to be valid. For example, if a wallet requires three confirmations to consider a transaction valid, it would require the next two blocks after the block including the transaction to confirm the transaction. 
+
+
+#### Consensus rules
+
+Rules, encoded in software, that allow a network of nodes to reach an agreement about the state of the ledger. Rules include what data blocks can contain, how block data is structured, and how nodes validate blocks. 
+
+#### Consensus rules voting
+
+Changes to Decred's consensus rules can only be made through an [on-chain voting process](https://docs.decred.org/getting-started/user-guides/agenda-voting/) which lasts for around one month. The change is made only if at least 75% of votes are in favor. 
+
+#### Constitution
+
+A [document](https://docs.decred.org/getting-started/constitution/) which defines the purpose and guiding principles of the Decred project.
+
+#### Credits
+
+The full unit of the Decred currency (i.e. 1 DCR).
+
+
+## D
+
+---
+
+#### DCR
+
+Ticker for the Decred currency.
+
+#### Decred Change Proposal (DCP)
+
+A [design document](https://github.com/decred/dcps) that describes potential protocol or consensus changes to Decred. DCPs primarily serve for documentation, fostering cross-implementation compatibility and helping ensure proper engineering rigor is followed.
+
+#### Decrediton
+
+A GUI (Graphical User Interface) [Decred wallet](https://github.com/decred/decrediton/) maintained by the core team. 
+
+#### Difficulty
+
+Difficulty is a measure of how difficult it is to mine a new block (i.e. find a hash below a given target). In Decred, the PoW difficulty is calculated from the exponentially weighted average of differences in previous block times.  
+
+#### Distributed ledger
+
+A cryptographically secure ledger, composed of nodes in a network, where each node holds a copy of the same ledger. 
+
+
+## E
+
+---
+
+#### Expired ticket
+
+Tickets that reached the end of their window without being called to vote - these can be revoked, but do not grant a reward.
+
+
+## F
+
+---
+
+#### Faucet
+
+A mechanism for obtaining free (testnet) tokens.
+
+
+## H
+
+---
+
+#### Hash
+
+The output of a cryptographic hashing function that produces a fixed-size and unique value from variable-size input. 
```

Comment:
Hashes are not unique, but good hash functions are astronomically difficult to reverse or to find collisions.

https://en.wikipedia.org/wiki/Pigeonhole_principle

Reply:
Interesting. Good to know. Updated definition below. I've also updated the 'Hash Function' definition accordingly (also below).

#### Hash

The output of a cryptographic hashing function that produces a fixed-size value from variable-size input. While it is computationally easy to create a hash from an input, it is extremely computationally difficult to calculate an input that will produce a given hash. 


#### Hash function

A cryptographic function that produces a fixed-size hash value from variable-size transaction input. Decred's Proof-of-Work uses the [BLAKE-256](https://docs.decred.org/research/blake-256-hash-function/) hashing function.